### PR TITLE
[ci-tools] add the sanitize-prow-jobs binary to the ci-operator-prowgen image

### PR DESCRIPTION
**Pull request by droslean**
/cc @openshift/test-platform @danilo-gemoli 

after https://github.com/openshift/ci-tools/pull/3614 and https://github.com/openshift/ci-tools/pull/3613
we need the binary to make sure that the `generate-config` job runs the sanitizer too.

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

File: ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml [hunk 1]
```diff
@@ -774,7 +776,8 @@ tests:
         name: hive-hive-credentials
         namespace: test-credentials
       from: test-bin
-      resources:
+      resources:ince the location of the binary is
```
Comment (danilo-gemoli) [inline ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml:779]: What's this line?

Comment (droslean) [reply ci-operator/config/openshift/ci-tools/openshift-ci-tools-master.yaml:779]: jez

**Pull request merged**